Include/object.h

```diff
@@ -624,12 +619,12 @@ static inline Py_ALWAYS_INLINE void Py_INCREF(PyObject *op)
     // directly PyObject.ob_refcnt.
 #if SIZEOF_VOID_P > 4
     // Portable saturated add, branching on the carry flag and set low bits
-    PY_UINT32_T cur_refcnt = op->ob_refcnt_split[PY_BIG_ENDIAN];
+    PY_UINT32_T cur_refcnt = _Py_CAST(PY_UINT32_T, op->ob_refcnt);
     PY_UINT32_T new_refcnt = cur_refcnt + 1;
     if (new_refcnt == 0) {
         return;
     }
-    op->ob_refcnt_split[PY_BIG_ENDIAN] = new_refcnt;
+    memcpy(&op->ob_refcnt, &new_refcnt, sizeof(new_refcnt));
```

Comment:
Isn't this require the little-endian?

Reply:
Maybe move the union here, use memcpy() and only use the bits that you need in the union?

Reply:
The problem of the current implementation (without this change) is the usage of *anonymous* union in PyObject: it is not compatible with (strict) C99. A *named* union is fine with C89 and C99. It's more about finding the right syntax to get the most efficient code.